test/express.ts

```diff
@@ -181,6 +182,63 @@ void tap.test(
 	}
 )
 
+void tap.test(
+	'maintains unique value when the library is depended upon both directly and transitively',
+	async (t) => {
+		const app = express()
+
+		// this function in the test library does app.use(middleware) and
+		// httpContext.set(REQUEST_ID_CONTEXT_KEY, <a unique id>)
+		init(app)
+
+		app.get('/', ((req, res) => {
+			set('value', req.query['value'])
+
+			res.status(200).json({
+				value: req.query['value'],
+				valueFromContext: get<number>('value'),
+				requestId: get<string>(REQUEST_ID_CONTEXT_KEY)
+			})
+		}))
+
+		const request = supertest(app)
+		const [response1, response2] = await Promise.all([
+			request.get('/').query({ value: 'value1' }),
+			request.get('/').query({ value: 'value2' }),
+		])
+
+		console.log(response1.body);
+		console.log(response1.headers);
+		console.log(response2.body);
+		console.log(response2.headers);
+
+		t.strictSame(response1.body.value, 'value1')
+		t.strictSame(response2.body.value, 'value2')
+
+		t.type(response1.header[REQUEST_ID_IN_RESPONSE_HTTP_HEADER_NAME], "string")
+		t.strictSame(response1.header[REQUEST_ID_IN_RESPONSE_HTTP_HEADER_NAME].length, 21)
+		t.type(response2.header[REQUEST_ID_IN_RESPONSE_HTTP_HEADER_NAME], "string")
+		t.strictSame(response2.header[REQUEST_ID_IN_RESPONSE_HTTP_HEADER_NAME].length, 21)
```

Comment:
These validations succeed - the library code setting the generated `nanoid` values into the response headers directly works just fine.